projects/fw_103/src/main.c

```diff
@@ -0,0 +1,59 @@
+#include <stdint.h>  // for integer types
+
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"  // interrupts are required for soft timers
+#include "log.h"        // for printing
+#include "wait.h"       // for wait function
+
+GpioAddress adc_addr = {
```

Comment:
when we have global variables like this that should only be visible within one file, we define them as static and prefix the variable name with s_{name}.

Reply:
Noted, thank you